### PR TITLE
fix(PasswordView): Space is not allowed in password

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do
Closing: #8385 
Update new password input validators to allow any character input and show error message if the character is invalid.
Add PasswordView to storybook

### Affected areas
PasswordView

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/47811206/b0f3d5ee-e903-4327-8e46-15f27f88ad21

- [x] I've checked the design and this PR matches it

### CODE REVIEW

File: ui/imports/shared/views/PasswordView.qml [hunk 1]
```diff
@@ -78,14 +76,26 @@ ColumnLayout {
         property bool containsNumbers: false
         property bool containsSymbols: false
 
-        readonly property var validator: RegExpValidator { regExp: /^[!-~]{0,64}$/ } // That incudes NOT extended ASCII printable characters less space and a maximum of 64 characters allowed
+        readonly property var validatorRegexp: /^[!-~]{0,64}$/
+        readonly property var validator: RegExpValidator { regExp: d.validatorRegexp } // That incudes NOT extended ASCII printable characters less space and a maximum of 64 characters allowed
+        readonly property string validatorErrMessage: qsTr("Only letters, numbers, underscores and hyphens allowed")
```
Comment (caybro) [inline ui/imports/shared/views/PasswordView.qml:81]: But we do allow spaces too, right? So this error message is a bit confusing

Comment (alexjba) [reply ui/imports/shared/views/PasswordView.qml:81]: The input field should accept space, but invalidate the password to let the user know space is not allowed.

Comment (alexjba) [reply ui/imports/shared/views/PasswordView.qml:81]: Sorry, forgot to link it to the bug. There's more info in there.
> Closing: https://github.com/status-im/status-desktop/issues/8385

Comment (caybro) [reply ui/imports/shared/views/PasswordView.qml:81]: Ah I see!

File: ui/imports/shared/views/PasswordView.qml [hunk 2]
```diff
@@ -78,14 +76,26 @@ ColumnLayout {
...
 
         // Password strength categorization / validation
         function lowerCaseValidator(text) { return (/[a-z]/.test(text)) }
         function upperCaseValidator(text) { return (/[A-Z]/.test(text)) }
         function numbersValidator(text) { return (/\d/.test(text)) }
         // That incudes NOT extended ASCII printable symbols less space:
         function symbolsValidator(text) { return (/[!-\/:-@[-`{-~]/.test(text)) }
+        function inputValidator(text) { return (validatorRegexp).test(text) }
```
Comment (caybro) [inline ui/imports/shared/views/PasswordView.qml:89]: You don't seem to be using this function at all, except inside `validateInput(text)`; remove it?

Comment (caybro) [reply ui/imports/shared/views/PasswordView.qml:89]: Also, the name suggests it returns some kind of validator when in fact in validates (tests) the input

Comment (alexjba) [reply ui/imports/shared/views/PasswordView.qml]: Done

**Review by caybro — APPROVED** [review]
LGTM!

**Review by noeliaSD — CHANGES_REQUESTED** [review]
Code LGTM! However, tested and I have 2 comments:

1. When the first introduced character is a space there is something wrong with the strengh bar:

https://github.com/status-im/status-desktop/assets/97019400/ba5d846c-39c5-40d2-b591-427ac14b8c60

2. Should the `Current password` input also allow to type spaces?

**Comment by alexjba** [comment]
> 1. When the first introduced character is a space there is something wrong with the strengh bar:

What is wrong with it? Can't spot it in the video🤔 

> 2\. Should the `Current password` input also allow to type spaces?

I didn't update the `Current password` input to allow spaces, but now I think I should have.
In the login view the user is allowed to type spaces in the password input field. I guess we should also allow the user to type spaces on current password input when changing the password. @benjthayer WDYT?

**Comment by noeliaSD** [comment]
> > 1. When the first introduced character is a space there is something wrong with the strengh bar:
> 
> What is wrong with it? Can't spot it in the video🤔 Is it about the increasing strength bar when there's a space in the password?
> 

The bar is black and should be red with some text.

**Comment by alexjba** [comment]
> > > 1. When the first introduced character is a space there is something wrong with the strengh bar:
> > 
> > 
> > What is wrong with it? Can't spot it in the video🤔 Is it about the increasing strength bar when there's a space in the password?
> 
> The bar is black and should be red with some text.

Ohh, it's black in storybook. In the app is red with some text. I guess we're missing some color definition. Will look on it

**Comment by noeliaSD** [comment]
> > > > 1. When the first introduced character is a space there is something wrong with the strengh bar:
> > > 
> > > 
> > > What is wrong with it? Can't spot it in the video🤔 Is it about the increasing strength bar when there's a space in the password?
> > 
> > 
> > The bar is black and should be red with some text.
> 
> Ohh, it's black in storybook. In the app is red with some text. I guess we're missing some color definition. Will look on it

I've also seen it in the app when you introduce a space in the first position 🤔

**Comment by alexjba** [comment]
> > > 1. When the first introduced character is a space there is something wrong with the strengh bar:
> > 
> > 
> > What is wrong with it? Can't spot it in the video🤔 Is it about the increasing strength bar when there's a space in the password?
> 
> The bar is black and should be red with some text.

Thanks for spotting that! 🍻 
Fixed both in the app and storybook

**Review by noeliaSD — APPROVED** [review]
Tested again and it works fine!

**Comment by alexjba** [comment]
Quoting @benjthayer 
> for the user's existing password, we just validate it as correct or incorrect, for the new password field we flag invalid characters, for the confirm password we simply say doesnt match new password

**Pull request merged**